lib/Parse/ParseDecl.cpp

```diff
@@ -2882,11 +2912,43 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
 
     consumeAttributeLParen();
 
+    if (Tok.is(tok::code_complete)) {
+      if (CodeCompletionCallbacks) {
+        CodeCompletionCallbacks->completeDeclAttrParam(
+            ParameterizedDeclAttributeKind::AccessControl, 0, false);
+      }
+      consumeToken(tok::code_complete);
+      return makeParserCodeCompletionStatus();
+    }
+    
     // Parse the subject.
     if (Tok.isContextualKeyword("set")) {
       consumeToken();
     } else {
       diagnose(Loc, diag::attr_access_expected_set, AttrName);
+      
+      const Token &Tok2 = peekToken();
+      
+      if (Tok.is(tok::code_complete)) {
+        if (CodeCompletionCallbacks) {
+          CodeCompletionCallbacks->completeDeclAttrParam(
+              ParameterizedDeclAttributeKind::AccessControl, 0, false);
+        }
+        consumeToken(tok::code_complete);
+        return makeParserCodeCompletionStatus();
+      }
```

Comment:
I don't think this is ever hit. If `Tok` is a code completion token here, then we would have already returned from the `Tok.is(too::code_complete)` check above.

Reply:
Nice catch, I guess I just accidentally duplicated that check 😅